src/SDKs/Azure.Core/data-plane/Azure.Core/Response.cs

```diff
@@ -14,7 +14,7 @@ public abstract class Response: IDisposable
 
         public abstract Stream ContentStream { get; set; }
 
-        public abstract string RequestId { get; set; }
+        public abstract string ClientRequestId { get; set; }
```

Comment:
Why is it here and not on ResponseHeaders?

Reply:
Because I want it to be flown from request and used for logging even if the server doesn't support echoing client request id back.

Reply:
I'm curious about that as well;  seems odd to have one single bit  of metadata here, unless I'm not thinking of a scenario.

Reply:
> Because I want it to be flown from request and used for logging even if the server doesn't support echoing client request id back.

How does keeping it here helps in flowing it from the request?

Reply:
We can just copy a property value from request to response object in transport avoiding the need to set a header.

Reply:
Why cannot the property in the headers struct simply call an internal property in response?

Reply:
Because then it would look like it's coming from headers while sometimes it doesn't. I think what I'm trying to say is while headers might be used to roundtrip the ClientRequestId it's not inherently connected to headers.

Reply:
If that's the argument, then it should not be on Response either.

Reply:
Are you saying it should be on `HttpPipelineMessage`?

Reply:
If it's on HttpPipelineMessage users would not be able to read it. What I am saying is that if you are willing to have it on Response, despite that it might not come from the response,, you should be willing to put it on response's headers.

Reply:
I don't see the connection. For me, it's pretty strange that you can get something via `response.Headers.ClientRequestId` but not via `response.Headers` enumeration or `response.Headers.TryGet`.

The other way around is slightly more logical, ClientRequestId is the property of response that connects it to request it just isn't part of the response that came via the wire.

Reply:
OK, I buy this argument. I forgot we have a way to enumerate the headers.